codec-http2/src/main/java/io/netty/handler/codec/http2/HpackUtil.java

```diff
@@ -65,6 +65,19 @@ static int equalsConstantTime(CharSequence s1, CharSequence s2) {
         return ConstantTimeUtils.equalsConstantTime(s1, s2);
     }
 
+    /**
+     * Compare two {@link CharSequence}.
+     * @param s1 the first value.
+     * @param s2 the second value.
+     * @return {@code false} if not equal. {@code true} if equal.
+     */
+    static boolean equalsVariableTime(CharSequence s1, CharSequence s2) {
+        if (s1 instanceof AsciiString && s2 instanceof AsciiString) {
+            return s1.equals(s2);
+        }
+        return String.valueOf(s1).equals(String.valueOf(s2));
```

Comment:
@carl-mastrangelo I wonder why we not just use:

```java
return AsciiString.contentEquals(s1, s2);
```

Reply:
Done